edb/pgsql/compiler/relgen.py

```diff
@@ -3483,6 +3483,45 @@ def process_set_as_exists_expr(
     return new_stmt_set_rvar(ir_set, stmt, ctx=ctx)
 
 
+@_special_case('std::json_object_pack')
+def process_set_as_json_object_pack(
+        ir_set: irast.Set, stmt: pgast.SelectStmt, *,
+        ctx: context.CompilerContextLevel) -> SetRVars:
+    expr = ir_set.expr
+    assert isinstance(expr, irast.FunctionCall)
+
+    with ctx.subrel() as subctx:
+        ir_arg = expr.args[0].expr
+
+        # compile IR to pgast that may be a ref
+        arg_ref = dispatch.compile(ir_arg, ctx=subctx)
+
+        # ensure that pgast ref is actually a value
+        arg_val = output.output_as_value(arg_ref, env=subctx.env)
+
+        # get first and the second fields of the tuple
+        keys = astutils.tuple_getattr(arg_val, ir_arg.typeref, '0')
+        values = astutils.tuple_getattr(arg_val, ir_arg.typeref, '1')
+
+        # construct the function call
+        set_expr = pgast.FuncCall(
+            name=("jsonb_object_agg",), args=[keys, values]
```

Comment:
`jsonb_object_agg` will return `NULL` when there are no input rows, but we need to return an empty json object. We can solve that with a COALESCE.

Reply:
Oh, I need to get attentive to that!

Reply:
This is one of those "classic edgedb bugs" that has shown up several times